capsules/src/qdec.rs

```diff
@@ -0,0 +1,113 @@
+//! Provides userspace access to the Qdec on a board.
+//!
+//! Usage
+//! -----
+//!
+//! ````
+//! let qdec = static_init!(
+//!     capsules::qdec::Qdec<'static>,
+//!     capsules::qdec::QdecInterface::new(&nrf52::qdec::QDEC,
+//!                                         kernel::Grant::create())
+//! );
+//! kernel::hil::QdecDriver.set_client(qdec);
+//! ````
+//!
+//! #Number of Qdecs
+//! The number of qdecs available depends on the
+//! board/microcontroller
+//!
+//! #Interrupt Spurred Readings versus Regular Readings
+//! An application can either enable interrupts to get the
+//! accumulation value or manually read it whenever it wants
+
+use crate::driver;
+use kernel::hil;
+use kernel::debug;
+use core::cell::Cell;
+use kernel::{AppId, Callback, ReturnCode, Driver, Grant};
+
+pub const DRIVER_NUM: usize = driver::NUM::Qdec as usize;
+
+/// This struct contains the resources necessary for the QdecInterface
+pub struct QdecInterface<'a> {
+    driver: &'a dyn hil::qdec::QdecDriver,
+    apps: Grant<App>,
+}
+
+#[derive(Default)]
+/// This struct contains the necessary fields for an app
+pub struct App {
+    callback: Option<Callback>,
+    subscribed: bool,
+}
+
+impl QdecInterface<'a> {
+    /// Create a new instance of the QdecInterface
+    pub fn new (
+        driver: &'a dyn  hil::qdec::QdecDriver,
+        grant: Grant<App>,
+    ) -> QdecInterface<'a> {
+        QdecInterface {
+            driver: driver,
+            apps: grant,
+        }
+    }
+
+    /// Configure callback
```

Comment:
You probably don't need doc comments for functions only used internally within the capsule